PyMca5/tests/ut/test_q_tiled_catalog_selector_dialog.py

```diff
@@ -71,17 +71,23 @@ def test_url_editing(qtbot: QtBot, dialog_model: TiledCatalogSelector):
     dialog.url_entry.editingFinished.emit()
     assert dialog.model.url == "New url"
 
-
-def test_url_edit_focus(
-    qapp: QApplication,
-    qtbot: QtBot,
-    dialog_model: TiledCatalogSelector,
-):
-    """Verify custom event filtering for url_entry widget."""
+def test_url_set_through_model(qtbot: QtBot, dialog_model: TiledCatalogSelector):
+    """Verify setting the model url connects properly."""
     dialog = QTiledCatalogSelectorDialog(model=dialog_model)
     dialog.show()
     qtbot.addWidget(dialog)
 
-    dialog_model.url_buffer = "Some text"
-    qapp.sendEvent(dialog.url_entry, QEvent(QEvent.Type.FocusIn))
-    assert dialog_model.url_buffer == ""
+    expected_url = "New URL"
+    dialog.model.url = expected_url
+
+    with patch.object(dialog_model, "client_from_url") as mock_client_constructor:
+        client = Mock()
+        client.uri = dialog_model.url
+        client.context.api_uri = ""
+        mock_client_constructor.return_value = client
+
+        dialog.connect_button.click()
+
+    assert expected_url in dialog.connection_label.text()
+
+    assert expected_url in dialog.connection_label.text()
```

Comment:
This assertion showed up twice.  Not sure if you meant to make a different assertion, or if it just got duplicated accidentally.

```suggestion
```